democracy/views/section.py

```diff
@@ -295,52 +291,64 @@ def update(self, instance, validated_data):
+                    image = SectionImage.objects.filter(Q(section=None) | (Q(section=self.instance))).get(pk=pk)
+                except SectionImage.DoesNotExist:
+                    raise ValidationError("No image with ID %s available in this section" % pk)
+            elif reference_id := image_data.get("reference_id"):
+                # Create a new SectionImage based on an existing image.
                 try:
-                    if self.instance is not None:
-                        image = self.instance.images.get(pk=pk)
-                    else:  # Creating a copy of the hearing
-                        image = SectionImage.objects.get(pk=pk)
-                        image.pk = None
+                    image = SectionImage.objects.get(pk=reference_id)
                 except SectionImage.DoesNotExist:
-                    raise ValidationError("The Section does not have an image with ID %s" % pk)
-                serializer_params["instance"] = image
+                    raise ValidationError("Image %s does not exist" % reference_id)
+                image.pk = None
 
-            serializer = SectionImageCreateUpdateSerializer(**serializer_params)
+            serializer = SectionImageCreateUpdateSerializer(data=image_data, instance=image)
             serializer.is_valid(raise_exception=True)
 
-            # save serializer in data so it can be used when handling the images
+            # save serializer in data, so it can be used when handling the images
             image_data["serializer"] = serializer
 
         return data
 
     def validate_files(self, data):
         for index, file_data in enumerate(data):
-            pk = file_data.get("id")
             file_data["ordering"] = index
-            serializer_params = {"data": file_data, "context": {"request": self.context["request"]}}
 
-            if pk:
+            file = None
+            # NOTE: You can only use either pk or reference_id, not both. If you use both, pk will be used.
+            # reference_id is meant for "save as new" type of situations. Not sure how it should behave if
+            # used in conjunction with pk.
+            if pk := file_data.get("id"):
+                # Use an existing SectionFile.
                 try:
-                    if self.instance is not None:
-                        # only allow orphan files or files within this section already
-                        file = SectionFile.objects.filter(Q(section=None) | (Q(section=self.instance))).get(pk=pk)
-                    else:  # Creating a copy of the hearing
-                        file = SectionFile.objects.get(pk=pk)
-                        file.pk = None
-                except SectionImage.DoesNotExist:
+                    # only allow orphan files or files within this section already
+                    file = SectionFile.objects.filter(Q(section=None) | (Q(section=self.instance))).get(pk=pk)
```

Comment:
A minor thing, but I think the extra parentheses that are wrapping `(Q(section=self.instance))` can be removed.